pymc/distributions/timeseries.py

```diff
@@ -33,6 +35,159 @@
+           Random walk mean
+        sigma: np.ndarray
+            Standard deviation of innovation (sigma > 0)
+        init: float
+            Initialization value for GaussianRandomWalk
+        steps: int
+            Length of random walk, must be greater than 1. Returned array will be of size+1 to
+            account as first value is initial value
+        size: int
+            The number of Random Walk time series generated
+
+        Returns
+        -------
+        np.ndarray
+        """
+
+        if steps is None or steps < 1:
+            raise ValueError("Steps must be None or greater than 0")
+
+        # If size is None then the returned series should be (1+steps,)
+        if size is None:
+            init_size = 1
+            steps_size = steps
+
+        # TODO: Ask Ricardo how size is known to be a tuple? Its int above
+        # If size is None then the returned series should be (size, 1+steps)
+        else:
+            init_size = (*size, 1)
+            steps_size = (*size, steps)
+
+        init_val = rng.normal(init, sigma, size=init_size)
+        steps = rng.normal(loc=mu, scale=sigma, size=steps_size)
+
+        grw = np.concatenate([init_val, steps], axis=-1)
+
+        return np.cumsum(grw, axis=-1)
+
+
+gaussianrandomwalk = GaussianRandomWalkRV()
+
+
+class GaussianRandomWalk(distribution.Continuous):
+    r"""Random Walk with Normal innovations
+
+
+    Notes
+    -----
+    init is currently drawn from a Normal distribution with the same sigma as the innovations
+
+    Parameters
+    ----------
+    mu: tensor
+        innovation drift, defaults to 0.0
+    sigma: tensor
+        sigma > 0, innovation standard deviation, defaults to 0.0
+    init: float
+        Mean value of initialization, defaults to 0.0
+    steps: int
+        Number of steps in Gaussian Random Walks
+    size: int
+        Number of independent Gaussian Random Walks
+    """
+
+    rv_op = gaussianrandomwalk
+
+    @classmethod
+    def dist(
+        cls,
+        mu: Optional[Union[np.ndarray, float]] = 0.0,
+        sigma: Optional[Union[np.ndarray, float]] = 1.0,
+        init: float = 0.0,
+        steps: int = 0,
+        *args,
+        **kwargs
+    ) -> RandomVariable:
+
+        params = [at.as_tensor_variable(param) for param in (mu, sigma, init, steps)]
+
+        return super().dist(params, **kwargs)
+
+    def logp(
+        value: at.Variable,
+        mu: at.Variable,
+        sigma: at.Variable,
+        init: at.Variable,
+        steps: at.Variable,
+    ) -> at.TensorVariable:
+        """Calculate log-probability of Gaussian Random Walk distribution at specified value.
+
+        Parameters
+        ----------
+        value: at.Variable,
+        mu: at.Variable,
+        sigma: at.Variable,
+        init: at.Variable,
+
+        Returns
+        -------
+        TensorVariable
+        """
+
+        # Calculate initialization logp
+        init_logp = logprob.logp(Normal.dist(init, sigma), value[0])
```

Comment:
I don't think we should model the probability of the initial point. Users may want to give this an arbitrary prior, like for mu and sigma. Adding a logp term of this may lead to double counting of the same value and sampling issues for example when a user does:

```python
with pm.Model() as m:
  init = pm.StudentT("init", 4, 0, 1)
  grw = pm.GaussianRandomWalk("grw", 0, 1, steps=5, init=init)
```

We would be double counting init, once with a StudentT logp and another with the Normal logp.